operations/research-and-development/engineering/infrastructure-engineering/reliability-manifesto.md

```diff
@@ -0,0 +1,123 @@
+
+**D-1** Infrastructure team should be involved early in the design process for applications that will run in the infrastructure systems. Read about the [Production Readiness Review](/operations/research-and-development/engineering/infrastructure-engineering/production-readiness-review.md)
+
+**D-2** All new services should go through the POC phase and findings should be presented in a team meeting, together with proposal, design architecture and implementation steps. Example reference [here](https://docs.google.com/document/d/1KZlemVB_75Xv4CDao5bcWpKml3fHbpcoAk_gWl8nFSY/edit).
+
+**D-3** We should avoid monolithic designs for custom services when possible. Microservice design should be the aim and services should be open sourced for community usage when they don’t expose private company information.
+ 
+**D-4** All services should be followed by architecture diagrams (we use Lucidchart) when possible and documentation. 
+
+**D-5** Always be mindful that services work in a different way at scale. Configurations that might work in local environments could have a major impact in production systems. For example, a possible abuse of the amount of database connections used. 
+
+**D-6** All services should support the ability to turn off/on on demand without affecting the rest of the infrastructure services.
+
+## Observability
+
+The observability manifesto should cover three main pillars:
+- Logs
+- Metrics
+- Tracing
+
+### Logs
+
+**O-1** Logs should include meaningful messages and stack trace should not be a requirement to understand the log message.
+
+**O-2** Service logs are enriched with metadata & attributes that differentiate the service (e.g service: boards).
+
+**O-3** Logging levels should be consistent across services and false errors should be avoided to enable generation of metrics, patterns and dashboards.
+
+**O-4** Logging format structure should be consistent across services and should be agreed across all teams. String and JSON formats should be used for all logs. 
+
+**O-5** Retention of 30 days for all critical and 7 days of all non-critical production system logs should be met. With exceptions, logs can be kept up to 1 year in S3 utilizing Glacier options.
+
+### Metrics
+**O-6** All services should export key metrics under /metrics path and should be enriched with metadata & attributes that differentiate the service (e.g service: playbooks).
+
+**O-7** All the application metrics should be exposed via 8067 port.
+
+**O-8** Application and service software version metrics of all applications should be exposed.
+
+**O-** The application and services should expose availability and latency metrics.
+
+**O-10** Raw metrics should be kept for 7 days and 5m/1h resolution metrics should be kept for 365 days for all production systems. 
+
+### Tracing (Placeholder)
+O-11 As we are moving more into microservices, request tracing should be implemented for each microservice to enable distributed request tracing analysis.
```

Comment:
Could you share some context or reference about moving to microservices? I'm specifically interested to know the scope/breakdown of app/services to be moved.

Reply:
This is mostly applicable for the Mattermost Cloud, I would say which we need to revisit it a bit. For example:
- CWS = microservice
- Provisioner = microservice
- Elrond is a microservice
- Database factory is a microservice

etc. etc.

Reply:
Got it, thanks!